xgcm/grid.py

```diff
@@ -178,13 +178,22 @@ def __init__(
             # TODO (Julius in #568) full hierarchy of conventions here
             # but override with any user-given options
 
+            # try sgrid parsing
+            if sgrid.assert_valid_sgrid(ds):
+                sgrid_ax_names = sgrid.get_all_axes(ds)
+                parsed_coords = {}
+                for ax_name in sgrid_ax_names:
+                    parsed_coords[ax_name] = sgrid.get_axis_positions_and_coords(
+                        ds, ax_name
+                    )
             # try comodo parsing
-            comodo_ax_names = comodo.get_all_axes(ds)
-            parsed_coords = {}
-            for ax_name in comodo_ax_names:
-                parsed_coords[ax_name] = comodo.get_axis_positions_and_coords(
-                    ds, ax_name
-                )
+            else:
+                comodo_ax_names = comodo.get_all_axes(ds)
+                parsed_coords = {}
+                for ax_name in comodo_ax_names:
+                    parsed_coords[ax_name] = comodo.get_axis_positions_and_coords(
+                        ds, ax_name
+                    )
```

Comment:
If we could refactor the comodo logic in the same way, that would be great. 

Happy to help with that. We can discuss this at the meeting.

Reply:
Made an effort to start on this in tandem with SGRID refactor.
Made changes in `grid.py` moving to `metadata_parsers.py` and left `comodo.py` alone for now.